components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -215,11 +222,10 @@ func (n *Nats) DeleteSubscription(sub *eventingv1alpha1.Subscription) error {
 			"subject", s.Subject,
 		)
 
-		if strings.HasPrefix(key, subKeyPrefix) {
+		if subKeyPrefix == createKymaSubscriptionNamespacedName(key, s).String() {
```

Comment:
We agreed to  add a unit test for this as part of this issue https://github.com/kyma-project/kyma/issues/13161